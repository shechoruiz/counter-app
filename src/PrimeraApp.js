import React from 'react';

const PrimeraApp = () => {
  const salud = 'Hola mundo'

  return (
    <>
      <h1>{salud}</h1>
      <h2>Mi primera aplicaci√≥n</h2>
    </>
  );
};

export default PrimeraApp;
