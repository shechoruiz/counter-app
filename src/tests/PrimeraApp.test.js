import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en <PrimeraApp />", () => {
  // Prueba realizada con Jest
  // test('1. Debe mostar el mensaje "Hola, soy Sergio"', () => {
  //   const saludo = "Hola, soy Sergio";
  //   // const wrapper = render(<PrimeraApp saludo={saludo} />);
  //   /* Se puede hacer una desestructuraci√≥n del wrapper anterior y obtener
  //     directamente el getByText */

  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);

  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  // Prueba realizada con Enzyme

  test("2. Debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola, soy Sergio";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
});
