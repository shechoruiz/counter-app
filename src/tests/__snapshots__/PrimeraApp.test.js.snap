// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> 2. Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Sergio
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
